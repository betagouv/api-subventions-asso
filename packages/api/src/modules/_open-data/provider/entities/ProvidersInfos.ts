import { Provider } from "../@types/Provider";

export default class ProvidersInfos {
    constructor(public api: Provider[], public raw: Provider[]) {}
}
