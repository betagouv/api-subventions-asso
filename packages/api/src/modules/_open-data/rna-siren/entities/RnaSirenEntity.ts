import { Rna, Siren } from "dto";

export default class RnaSiren {
    constructor(public rna: Rna, public siren: Siren, public names?: string[]) {}
}
