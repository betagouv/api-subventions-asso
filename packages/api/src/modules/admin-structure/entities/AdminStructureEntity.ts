import { AgentTypeEnum } from "dto";

export default class AdminStructureEntity {
    constructor(public agentType: AgentTypeEnum, public territoryScope: string, public structure: string) {}
}
