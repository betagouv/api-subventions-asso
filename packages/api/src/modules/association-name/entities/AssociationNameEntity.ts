import { Adresse } from "dto";
import Rna from "../../../valueObjects/Rna";
import Siren from "../../../valueObjects/Siren";

export default class AssociationNameEntity {
    constructor(
        public name: string,
        public siren: Siren,
        public rna?: Rna,
        public address?: Adresse,
        public nbEtabs?: number,
    ) {}
}
