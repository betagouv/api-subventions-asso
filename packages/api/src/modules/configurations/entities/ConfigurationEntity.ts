export default interface ConfigurationEntity<T = unknown> {
    name: string;
    updatedAt: Date;
    data: T;
}
