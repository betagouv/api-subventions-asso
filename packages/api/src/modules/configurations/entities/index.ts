export * from "./DauphinTokenAvailableTimeDataEntity";
export * from "./DauphinTokenDataEntity";
export * as ConfigurationEntity from "./ConfigurationEntity";
