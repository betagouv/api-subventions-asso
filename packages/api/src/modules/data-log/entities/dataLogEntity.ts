export type DataLogEntity = {
    providerId: string;
    integrationDate: Date; // when we imported data
    editionDate?: Date; // date of file production or up to which date the file covers
    fileName?: string;
};
