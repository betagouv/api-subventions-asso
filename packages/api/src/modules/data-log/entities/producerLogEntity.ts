export type ProducerLogEntity = {
    providerId: string;
    lastIntegrationDate: Date; // when we last imported data
    firstIntegrationDate: Date; // when we first imported data
    lastCoverDate: Date; // last date any file had cover
};
