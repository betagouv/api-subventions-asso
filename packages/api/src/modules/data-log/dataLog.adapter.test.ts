import { DataLogAdapter } from "./dataLog.adapter";
import { ProducerLogEntity } from "./entities/producerLogEntity";

describe("DataLog Adapter", () => {
    describe("overviewToDto", () => {
        it("should return dto", () => {
            const entity: ProducerLogEntity = {
                providerId: "providerId",
                lastIntegrationDate: new Date("2024-09-11"),
                firstIntegrationDate: new Date("2018-04-23"),
                lastCoverDate: new Date("2024-09-07"),
            };

            const actual = DataLogAdapter.overviewToDto(entity);
            expect(actual).toMatchSnapshot();
        });
    });
});
