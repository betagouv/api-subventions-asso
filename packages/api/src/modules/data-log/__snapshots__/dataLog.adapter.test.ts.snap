// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataLog Adapter overviewToDto should return dto 1`] = `
Object {
  "derniere_date_edition": 2024-09-07T00:00:00.000Z,
  "derniere_date_integration": 2024-09-11T00:00:00.000Z,
  "identifiant_fournisseur": "providerId",
  "premiere_date_integration": 2018-04-23T00:00:00.000Z,
}
`;
